Python/getargs.c

```diff
@@ -2549,28 +2549,35 @@ _PyArg_UnpackKeywordsWithVararg(PyObject *const *args, Py_ssize_t nargs,
         return NULL;
     }
 
-    /* create varargs tuple */
+    /*
+     * Create the tuple of variadic arguments and place it at 'buf[vararg]'.
+     * The function has at most 'maxpos' positional arguments, possibly with
+     * default values or not and possibly positional-only. Since 'args' is
+     * an array of 'nargs' arguments representing the positional arguments
+     * of the call, there are only 'nargs - maxpos' arguments that will
+     * be considered as variadic and gathered into a single tuple.
+     */
     varargssize = nargs - maxpos;
     if (varargssize < 0) {
         varargssize = 0;
     }
-    buf[vararg] = PyTuple_New(varargssize);
-    if (!buf[vararg]) {
-        return NULL;
-    }
 
-    /* copy tuple args */
-    for (i = 0; i < nargs; i++) {
-        if (i >= vararg) {
-            PyTuple_SET_ITEM(buf[vararg], i - vararg, Py_NewRef(args[i]));
-            continue;
-        }
-        else {
-            buf[i] = args[i];
-        }
-    }
+    // copy the first arguments until the 'vararg' index
+    memcpy(buf, args, vararg * sizeof(PyObject *));
```

Comment:
Right and this is something I didn't check. I need to add tests for that (I would like to suggest a way to generate more clinic tests because there is a lack of coverage).